app/controllers/users/settings/account/connected_accounts_controller.rb

```diff
@@ -9,7 +9,8 @@ def index
         end
 
         def destroy
-          if Rails.configuration.x.azure_ad_apps.select { |_, v| v[:provider] == params[:provider] }.present?
+          settings = ApplicationSettings.instance
```

Comment:
Might be worth extracting settings into a helper method in the controller, to avoid constantly assigning it in controller an view.
```
def settings
  ApplicationSettings.instance
end
helper_method :settings
```
Unless we want to be intentionally explicit in each of these lines.